sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -806,6 +807,18 @@ object CollapseRepartition extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Substitute the aggregate expression which uses [[First]] as the aggregate function
```

Comment:
nit: `Replaces first(col) to nth_value(col, 1) for better performance.`

Reply:
OK